sqlness/src/interceptor/sort_result.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use std::collections::VecDeque;
+
+use crate::interceptor::{Interceptor, InterceptorFactory, InterceptorRef};
+
+const PREFIX: &str = "SORT_RESULT";
+
+/// Sort the query result in lexicographical order.
+///
+/// Grammar:
+/// ``` text
+/// -- SQLNESS SORT_RESULT <ignore-head> <ignore-tail>
+/// ```
+///
+/// Both `ignore-head` and `ignore-tail` are optional. Default value is 0 (no lines will be ignored).
+///
+/// # Example
+/// `.sql` file:
+/// ``` sql
+/// -- SQLNESS SORT_RESULT
+/// SELECT * from values (3), (2), (1);
+/// ```
+///
+/// `.result` file:
+/// ``` sql
+/// -- SQLNESS SORT_RESULT
+/// SELECT * from values (3), (2), (1);
+///
+/// 1
+/// 2
+/// 3
+/// ```
+#[derive(Debug)]
+pub struct SortResultInterceptor {
+    /// How much lines to ignore from the head
+    ignore_head: usize,
+    /// How much lines to ignore from the tail
+    ignore_tail: usize,
+}
+
+impl Interceptor for SortResultInterceptor {
+    fn after_execute(&self, result: &mut String) {
+        let mut lines = result.lines().collect::<VecDeque<_>>();
+        let mut head = Vec::with_capacity(self.ignore_head);
+        let mut tail = Vec::with_capacity(self.ignore_tail);
+
+        // ignore head and tail
+        for _ in 0..self.ignore_head {
+            if let Some(l) = lines.pop_front() {
+                head.push(l);
+            }
+        }
+        for _ in 0..self.ignore_tail {
+            if let Some(l) = lines.pop_back() {
+                tail.push(l);
+            }
+        }
+        tail.reverse();
+
+        // sort remaining lines
+        lines.make_contiguous().sort();
+
+        let new_lines = head
+            .into_iter()
+            .chain(lines.into_iter())
+            .chain(tail.into_iter())
+            .collect::<Vec<_>>();
+        *result = new_lines.join("\n");
+    }
+}
+
+pub struct SortResultInterceptorFactory;
+
+impl InterceptorFactory for SortResultInterceptorFactory {
+    fn try_new(&self, interceptor: &str) -> Option<InterceptorRef> {
```

Comment:
Why this method return Optione before? it seems Result is more reasonable, otherwise we cannot know why an interceptor isn't working as expected.

Maybe we can file an issue first.